### PR TITLE
Multisite 555

### PR DESCRIPTION
**Pull request by daovist**
#555 

This PR uses 4 new configuration options: closedRegistration, serveOnlyApproved, publishOnlyApproved, approvedChannels

There is no option to not show Publish in the navbar as described in the issue. Publish and Login are hidden based on the options defined

It's also not "disable publishing for non-authenticated users" but "only allow publishes from whitelisted channels/users"

### CODE REVIEW

File: client/src/reducers/site.js [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 
 let initialState = {
-  description       : 'default description',
-  googleAnalyticsId : 'default google id',
-  host              : 'default host',
-  title             : 'default title',
-  twitter           : 'default twitter',
-  defaultDescription: 'default description',
-  defaultThumbnail  : 'default thumbnail',
+  description        : 'default description',
```
[the remark this answers is not in the document]
Comment (kauffj) [reply client/src/reducers/site.js]: https://github.com/lbryio/spee.ch/issues/606

File: server/utils/isApprovedChannel.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const { publishing: { approvedChannels } } = require('@config/siteConfig');
```
[the remark this answers is not in the document]
Comment (daovist) [reply server/utils/isApprovedChannel.js]: I'm trying to learn best practices for these tools as I go so I'm not exactly sure, but I'm moving this to the utils folder shared by server and client, making it a pure function, and passing in `approvedChannels` from the config file.

File: client/src/reducers/site.js [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
+  googleAnalyticsId  : 'default google id',
+  host               : 'default host',
+  title              : 'default title',
+  twitter            : 'default twitter',
+  defaultDescription : 'default description',
+  defaultThumbnail   : 'default thumbnail',
+  closedRegistration : false,
+  serveOnlyApproved  : false,
+  publishOnlyApproved: false,
+  approvedChannels   : [],
```
[the remark this answers is not in the document]
Comment (daovist) [reply client/src/reducers/site.js]: This was a small amount of extra effort now to account for other configurations e.g. a spee.ch instance that serves all content but only publishes selected channels.

File: client/src/reducers/site.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (daovist) [reply client/src/reducers/site.js]: The default config is on the server. I think having silly things like "default thumbnail" are useful in development which is the only time these values should ever render.

File: server/speechPassport/utils/local-signup.js [hunk 4]
```diff
@@ -2,17 +2,21 @@ const PassportLocalStrategy = require('passport-local').Strategy;
...
+const { publishing: { closedRegistration } } = require('@config/siteConfig');
 
 module.exports = new PassportLocalStrategy(
   {
     usernameField: 'username',
     passwordField: 'password',
   },
   (username, password, done) => {
+    if (closedRegistration) {
+      return done('Registration is disabled');
```
[the remark this answers is not in the document]
Comment (daovist) [reply server/speechPassport/utils/local-signup.js]: That would have the same result and I can't find a way to make it any cleaner/simpler than this.

File: client/src/containers/NavigationLinks/view.jsx [hunk 5]
```diff
@@ -28,32 +29,33 @@ class NavigationLinks extends React.Component {
     }
   }
   render () {
+    const { site, channelLongId, channelName } = this.props;
     return (
       <div className='navigation-links'>
-        <NavLink
+        {(!site.publishOnlyApproved || isApprovedChannel({ longId: channelLongId }, site.approvedChannels)) && <NavLink
```
[the remark this answers is not in the document]
Comment (daovist) [reply client/src/containers/NavigationLinks/view.jsx]: moved to index.js

**Pull request merged**